### ISSUE TITLE
docker run with mitmproxy is finicky

### ISSUE THREAD

**Issue by cortesi** [mitmproxy/mitmproxy#1157]
##### Steps to reproduce the problem:
1. docker run mitmproxy/mitmproxy:dev
2. docker run -i mitmproy/mitmproxy:dev
##### What is the expected behavior?

The user should see some reasonable error output in both cases - when we attempt to run in a non-terminal, and when we're running in both a non-termina and non-interactively. 
##### What went wrong?

We see a crash in case 1) and an unresponsive screen in case 2.

**Comment by cortesi** [comment]
The correct usage here is of course:

    docker run -ti 

But we can expect users to make this error and they shouldn't be baffled when they do.

**Comment by javabrett** [comment]
Cases 1 & 2 should be do-able, but Case 3:

3. `docker run -t mitmproxy/mitmproxy:dev`

... looks to be harder.  Result is perhaps the worst of the three - mitmproxy UI displays but is non-responsive to keystrokes.  For 1 & 2 you can use `isatty()`, but I looked and am yet to find a way to detect that the host `stdin` is not connected to container mitmproxy process - which is presumably then connected to a tty which will never produce any input.  I looked at the fd link, its target - the `-i` case was indistinguishable.  Answers on SO etc. seem to only deal properly with detecting the tty.

For that case it might be necessary to 1) pass a running-in-docker arg and use it to 2) display an initial message in the UI to warn users if they have not `docker run` with `-it` then mitmproxy will appear unresponsive.

**Comment by Kriechi** [comment]
Fixed by fd633865937e596a647c1ad39a30d88b10a766c3
Unfortunately there is no way to detect a missing `-i` option, but I would argue that 99% of the search results always mention **just add -it and it will work**.

Now it behaves like this, including a new error message:
```
$ docker run mitmproxy/mitmproxy:dev                                                                                              [0]
Unable to find image 'mitmproxy/mitmproxy:dev' locally
dev: Pulling from mitmproxy/mitmproxy
6f821164d5b7: Pull complete
771446665261: Pull complete
f5dc80f69993: Pull complete
ea98edb07673: Pull complete
Digest: sha256:23df9012c64a8c48f686c97ddc844016fb8a9c6c81395853e533deb2eeb3bcdd
Status: Downloaded newer image for mitmproxy/mitmproxy:dev
Error: mitmproxy's console interface requires a tty. Please run mitmproxy in an interactive shell environment.
$ 
```
